src/components/views/CinemaHall.vue

```diff
@@ -0,0 +1,122 @@
+<script>
+import { defineComponent, ref } from 'vue';
+
+export default defineComponent({
+  props: {
+    rows: {
+      type: Number,
+      required: true,
+    },
+    seats: {
+      type: Number,
+      required: true,
+    },
+    seatsTaken: {
+      type: Array,
+      required: true
+    }
+  },
+  emits: ["seats-selected"],
+  setup(props, context) {
+
+    const rowLetter = (rowNumber) => {
+      return String.fromCharCode(96 + rowNumber).toUpperCase();
+    }
+
+    const isSeatTaken = (seat) => {
+      return props.seatsTaken.includes(seat)
+    }
+
+    const seatsSelected = ref([]);
+
+    const handleClick = (e) => {
+      e.preventDefault();
+      if (!props.seatsTaken.includes(e.target.value)) {
+        if (seatsSelected.value.includes(e.target.value)) {
+          seatsSelected.value = seatsSelected.value.filter(item => item !== e.target.value);
+        }
+        else {
+          seatsSelected.value.push(e.target.value);
+        }
+        context.emit('seats-selected', { seatsSelected: seatsSelected.value})
```

Comment:
Is there any advantage to use here `v-model` instead of a custom event? 😺

Reply:
not exactly - you can even still use just an event but it gives you convention how to name it and suggests that it is there to bind some value between components.  The thing is that with v-model you can give the responsibility of storing/updating a value to the parent and it's pretty useful for form components (it may not be obvious at first sight but seat picker is actually just a simple form element 😉).

But as I said it's definitely a minor thing (that may pay off in the future).

Reply:
Wow, this is a real eye-opener! 😃 It is a kind of form indeed!